aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConfiguration.java

```diff
@@ -58,6 +70,76 @@ public static Builder builder() {
         return new Builder();
     }
 
+    /**
+     *
+     * @param context application context which will be used to retrieve amplicationconfiguration.json
+     * @param pluginConfigKey the desired dataStore plugin config key (i.e. awsDataStorePlugin)
+     * @return A new builder instance
+     * @throws DataStoreException
+     */
+    @NonNull
+    public static Builder builder(@NonNull Context context, @NonNull String pluginConfigKey) throws DataStoreException {
+        Objects.requireNonNull(context);
+        Objects.requireNonNull(pluginConfigKey);
+        JSONObject pluginConfig = null;
+        try {
+            pluginConfig = Objects.requireNonNull(AmplifyConfiguration.fromConfigFile(context)
+                .forCategoryType(CategoryType.DATASTORE))
+                .getPluginConfig(pluginConfigKey);
+        } catch (AmplifyException | NullPointerException exception) {
+            Log.w(TAG, "Unable to read DataStore configuration from file.", exception);
+        }
+        return pluginConfig != null ? builder(pluginConfig) : builder();
+    }
+
+    /**
+     * Begin building a new instance of {@link DataStoreConfiguration}.
+     * @param pluginJson The dataStore configuration as a JSONObject
+     * @return A new builder instance
+     * @throws DataStoreException exception thrown if there's an unexpected configuration key or
+     * an invalid configuration value
+     */
+    @NonNull
+    public static Builder builder(@NonNull JSONObject pluginJson) throws DataStoreException {
+        final Iterator<String> jsonKeys = pluginJson.keys();
+        Builder builder = new Builder();
+        while (jsonKeys.hasNext()) {
+            final String keyString = jsonKeys.next();
+            final ConfigKey configKey;
+            try {
+                configKey = ConfigKey.fromString(keyString);
+            } catch (IllegalArgumentException noSuchConfigKey) {
+                throw new DataStoreException(
+                        "Saw unexpected config key: " + keyString,
+                        "Make sure your amplifyconfiguration.json is valid."
+                );
+            }
+
+            try {
+                switch (configKey) {
+                    case SYNC_INTERVAL:
```

Comment:
iOS appears to be using milliseconds while JS uses minutes. Probably something we need to align on. @drochetti @wooj2 @palpatim @jamesonwilliams 

I don't have a super strong opinion one way or another, but milliseconds does leave the door open for some potential chattiness.

Reply:
on Swift we follow the platform standard of representing the time intervals with `Double`. However, there's an idiomatic way of doing that, so our actual configuration code looks like this:

```swift
.custom(
    syncTimeInterval: .minutes(10)
)
```

Therefore the code itself is explicit about the time unit being used. Behind the scenes the Double represents the time in **seconds**. But since it's a Double, milliseconds can be achieved by using decimal points (or through `.milliseconds(500)`). At least the intent is explicit.

https://github.com/aws-amplify/amplify-ios/blob/master/AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/DataStoreConfiguration.swift#L76

Reply:
Would it be reasonable for Android to use minutes?

Reply:
We could modify the `DataStoreConfiguration` constructor and rename the `syncIntervalMs` to `syncInterval` and the do the following in the constructor body:

```
this.syncInterval = TimeUnit.MINUTES.toMillis(syncInterval);
```

Thoughts?

Reply:
@rjuliano that was our path on iOS. We respected the platform default, of expressing seconds as a `Double` and make sure we have an API to clearly express the duration unit (without the `Ms` suffix).

My 2 cents is that maybe a wrapper around `TimeUnit` could be a good way to express the API that would work well in both Java and Kotlin (and very similar to the one in Swift):

```java
import static com.amplifyframework.util.TimeUnitUtils.minutes; // Utils for a lack of better word now

Long syncInterval = minutes(5); // delegates to TimeUnit.MINUTES.toMillis(5);
```